pkg/proxy/chat.go

```diff
@@ -0,0 +1,170 @@
+		logger.Errorf("Get account chat info error: %s", err)
+		return nil, err
+	}
+
+	id := common.UUID()
+
+	apiSession := &model.Session{
+		ID:         id,
+		User:       user,
+		LoginFrom:  model.LoginFromWeb,
+		RemoteAddr: ip,
+		Protocol:   model.ActionALL,
+		Asset:      accountInfo.Asset.Name,
+		Account:    accountInfo.Name,
+		AccountID:  accountInfo.ID,
+		AssetID:    accountInfo.Asset.ID,
+		UserID:     userID,
+		OrgID:      "00000000-0000-0000-0000-000000000004",
+		Type:       model.NORMALType,
+		LangCode:   langCode,
+		DateStart:  modelCommon.NewNowUTCTime(),
+	}
+
+	_, err2 := jmsService.CreateSession(*apiSession)
+	if err2 != nil {
+		return nil, err2
+	}
+
+	chat := &ChatJMSServer{
+		JmsService: jmsService,
+		Session:    apiSession,
+		Conf:       conf,
+	}
+
+	chat.CmdR = chat.GetCommandRecorder()
+	chat.Replay = chat.GetReplayRecorder()
+
+	if err1 := jmsService.RecordSessionLifecycleLog(id, model.AssetConnectSuccess,
+		model.EmptyLifecycleLog); err1 != nil {
+		logger.Errorf("Record session activity log err: %s", err1)
+	}
+
+	return chat, nil
+}
+
+type ChatJMSServer struct {
+	JmsService *service.JMService
+	Session    *model.Session
+	CmdR       *CommandRecorder
+	Replay     *ChatReplyRecorder
+	Conf       *model.TerminalConfig
+}
+
+func (s *ChatJMSServer) GenerateCommandItem(user, input, output string) *model.Command {
+	createdDate := time.Now()
+	return &model.Command{
+		SessionID:   s.Session.ID,
+		OrgID:       s.Session.OrgID,
+		Input:       input,
+		Output:      output,
+		User:        user,
+		Server:      s.Session.Asset,
+		Account:     s.Session.Account,
+		Timestamp:   createdDate.Unix(),
+		RiskLevel:   model.NormalLevel,
+		DateCreated: createdDate.UTC(),
+	}
+}
+
+func (s *ChatJMSServer) GetReplayRecorder() *ChatReplyRecorder {
+	info := &ReplyInfo{
+		Width:     200,
+		Height:    200,
+		TimeStamp: time.Now(),
+	}
+	recorder, err := NewReplayRecord(s.Session.ID, s.JmsService,
+		NewReplayStorage(s.JmsService, s.Conf),
+		info)
+	if err != nil {
+		logger.Error(err)
+	}
+
+	return &ChatReplyRecorder{recorder}
+}
+
+func (s *ChatJMSServer) GetCommandRecorder() *CommandRecorder {
+	cmdR := CommandRecorder{
+		sessionID:  s.Session.ID,
+		storage:    NewCommandStorage(s.JmsService, s.Conf),
+		queue:      make(chan *model.Command, 10),
+		closed:     make(chan struct{}),
+		jmsService: s.JmsService,
+	}
+	go cmdR.record()
+	return &cmdR
+}
+
+func (s *ChatJMSServer) Close(msg string) {
+	session.RemoveSessionById(s.Session.ID)
+	if err := s.JmsService.SessionFinished(s.Session.ID, modelCommon.NewNowUTCTime()); err != nil {
+		logger.Errorf("finish session %s: %v", s.Session.ID, err)
+	}
+
+	s.CmdR.End()
+	s.Replay.End()
+
+	logObj := model.SessionLifecycleLog{Reason: msg, User: s.Session.User}
+	err := s.JmsService.RecordSessionLifecycleLog(s.Session.ID, model.AssetConnectFinished, logObj)
+	if err != nil {
+		logger.Errorf("record session lifecycle log %s: %v", s.Session.ID, err)
+		return
+	}
+}
```

Comment:
The code has no significant problems or inconsistencies that need to be addressed at this point in time.

There is some minor formatting issues like inconsistent spacing around opening brackets, which I will fix:

```go
func (rh *ChatReplyRecorder) WriteInput(inputStr string) {
currentTime := time.Now()
formattedTime := currentTime.Format("2006-01-02 15:04:05")
inputStr = fmt.Sprintf("[%s]#: %s", 
```

Please review the documentation if you find additional changes needed to align with community standards and conventions.